civihr_employee_portal/views/includes/civihr_employee_portal_handler_absence_table_header_block_area.inc

```diff
@@ -0,0 +1,212 @@
+<?php
+
+/**
+ * This class contains View Handler for overriding absence_list view.
+ * It's used to generate Absence List custom table structure based
+ * on Drupal View's fields settings.
+ */
+class civihr_employee_portal_handler_absence_table_header_block_area extends views_handler_area {
+
+  /**
+   * Render the area.
+   */
+  public function render($empty = FALSE) {
+
+    $header = $this->getHeaderRow();
+    $rows = array();
+
+    $periodId = isset($_REQUEST['absence_start_date_period_filter']) ?: variable_get('default_date_period_id', '1');
+    $entitlements = $this->getEntitlementData($periodId);
+    $rows[] = $this->getEntitlementRow($entitlements);
+    $rows = array_merge($rows, $this->getContentRows($entitlements));
+
+    return theme('table', array('header' => $header, 'rows' => $rows));
+  }
+
+  /**
+   * Return an array containing header row.
+   * 
+   * @return array
+   */
+  private function getHeaderRow() {
+
+    $row = array();
+    $absenceTypes = $this->getAbsenceTypes();
+
+    $row[] = t('Description');
+    $row[] = t('Status');
+    foreach ($absenceTypes as $absenceType) {
+      $row[] = $absenceType['title'];
+    }
+
+    return $row;
+  }
+
+  /**
+   * Return an array containing entitlements row for given $entitlements array.
+   *
+   * @param array $entitlements
+   * @return array
+   */
+  private function getEntitlementRow(array $entitlements) {
```

Comment:
I think this is more formatting rather than actually getting the entitlements .. so maybe formatEntitlementRow is a better name ?